### ISSUE TITLE
Improve CLI input overrides

### ISSUE THREAD

**Issue by RunDevelopment** [chaiNNer-org/chaiNNer#1496]
With #1494, we now have a system to overrides input within a chain before CLI execution. However, the current solution has some drawbacks.

Before I explain other solutions, let's look at the problem. To make the CLI `run` command more useful, I added the `--override` option, which allows users to set certain inputs inside the chain before it is run. Conceptually, users need to tell chainner which input to override and what value to set it to.

### The current solution

Currently, CLI input overrides are defined in a JSON file of the following format:
```json
{
  "inputs": {
    "<override input id>": <value>
  }
}
```
`<override input id>` is a combination of the node id (UUID) and input id (number) to override. `<value>` is either a string, number, or null. Since the override input id contains the node id and input id, it easy to identify a specific input within a chain. The value will then be assigned directly to the `inputData` of the node (after some validation).

the override input id itself is obtained in the GUI:
![image](https://user-images.githubusercontent.com/20878432/213469740-eba96539-53e5-4a4f-a147-810f7a9db207.png)

This approach has 2 main limitations:
1. It does not support all inputs. Only file, directory, text, number, and slider inputs are supported. Dropdowns are not supported. This is because users can't know the underlying values dropdowns use. Dropdowns typically use (seemly) random numbers/strings to represent the current option, so overrides would have to also use those numbers/strings.
2. Override input ids are specific to a certain chain. Since an override input id directly contains the node id, it can only override the inputs of that node. This means that e.g. `"<override input id>": 42` overrides the value of a specific node in a specific chain. So you can't reuse overrides across chains.

### CLI input node

Joey also had another idea on how to implement overrides. He suggested a CLI input node (prototype image below). Basically, the node in the GUI would just be an input + a name. The input specifies the default value and the name would be used to reference it in the override file. E.g. in the screenshot before, the override could look like this: `"inputs": { "FileName": "the override value" }`.

![image](https://user-images.githubusercontent.com/20878432/213464970-c2550a0f-36bb-4e7d-8cd4-407d79389109.png)

While this approach does solve problem (2), it also some limitations: 
- You can only override connectable inputs. Most file and some text input do not allow connections for technical reasons.
- This does not support optional inputs. We don't support optional outputs right now, so the CLI input node always has to output *some* value.

### External names

Another idea I recently had is to give nodes external names. Basically, the UI would give users an option to give nodes a special name with which users can refer to that node in override files. This would allow users to reuse override files across chains as long as they use the same external names.

In the UI, external names could look like this:
![image](https://user-images.githubusercontent.com/20878432/213469115-42d09acf-3d39-472f-8c30-4a47b719b343.png)

When the user then right clicks to get the override input id, instead of `#7e857dd9-39ae-4cc3-8430-eb5aab3a6986:1`, they'll get `FileName:1`.

However, it also doesn't completely solve the problem (2), because the external name only idenifies a node, but not an input. So the external name would "only" make override input ids more readable and reusable, but it would still have to contain the random id of the input itself.

**Comment by joeyballentine** [comment]
I'd rather not have something super specific to CLI features be present on nodes all the time. That's just adding clutter for a feature that 99% of users will never use.

**Comment by RunDevelopment** [comment]
Agreed. With external names, I thought of them as something that would be added via the context menu. So nodes don't display anything until an external name is given to them.
